File: Medium/1456. Maximum Number of Vowels in a Substring of Given Length/Solution.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int MaxVowels(string s, int k) {
        var vowelSet = new HashSet<char> { 'a', 'e', 'i', 'o', 'u' };
        
        int currentVowels = 0;
        
        // İlk pencereyi hesapla
        for (int i = 0; i < k; i++) {
            if (vowelSet.Contains(s[i])) {
                currentVowels++;
            }
        }
        
        int maxVowels = currentVowels;
        if (maxVowels == k) return k;
        
        // Pencereyi kaydır
        for (int i = k; i < s.Length; i++) {
            // Yeni giren karakter
            if (vowelSet.Contains(s[i])) {
                currentVowels++;
            }
            
            // Pencereden çıkan karakter
            if (vowelSet.Contains(s[i - k])) {
                currentVowels--;
            }
            
            maxVowels = Math.Max(maxVowels, currentVowels);
            if (maxVowels == k) return k;
        }
        
        return maxVowels;
    }
}

File: Medium/1456. Maximum Number of Vowels in a Substring of Given Length/Solution.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = {'a', 'e', 'i', 'o', 'u'}
        current_vowels = 0
        
        for i in range(k):
            if s[i] in vowels:
                current_vowels += 1
                
        max_vowels = current_vowels
        if max_vowels == k:
            return k
            
        for i in range(k, len(s)):
            if s[i] in vowels:
                current_vowels += 1
            
            if s[i - k] in vowels:
                current_vowels -= 1
                
            max_vowels = max(max_vowels, current_vowels)
            
            if max_vowels == k:
                return k
                
        return max_vowels